docs/source/distributed.fsdp.fully_shard.rst

```diff
@@ -0,0 +1,54 @@
+torch.distributed.fsdp.fully_shard
+==================================
+
+PyTorch FSDP2 (``fully_shard``)
+-------------------------------
+
+PyTorch FSDP2 provides a fully sharded data parallelism (FSDP) implementation
+targeting performant eager-mode while using per-parameter sharding for improved
+usability. Compared to PyTorch FSDP1 (``FullyShardedDataParallel``):
+
+- FSDP2 uses ``DTensor``-based dim-0 per-parameter sharding for a simpler
+  sharding representation compared to FSDP1's flat-parameter sharding, while
+  preserving similar throughput performance. This provides a more intuitive
+  user experience, relaxes constraints around frozen parameters, and allows for
+  communication-free sharded state dicts.
```

Comment:
Should we also add that `full state dict` is not supported by FSDP2?

Reply:
Added a bullet